ui/src/services/utils.js

```diff
@@ -285,7 +287,10 @@ export function addMissingDataPoint(
     orginalValues.length === 0 ||
     !startingTimeStamp ||
     !sampleDuration ||
-    !sampleFrequency
+    !sampleFrequency ||
+    // if the `sampleDuration` and `sampleFrequency` is not from the predefined value
+    !SAMPLE_DURATION_LIST.includes(sampleDuration) ||
+    !SAMPLE_FREQUENCY_LIST.includes(sampleFrequency)
```

Comment:
Why add this behaviour? It only seems an artificial limitation to me (not showing an option in the visible UI doesn't mean your utility function can't do something).

Reply:
For me, this function is **only** useful for the graph. And the choices of `sampleDuration` and `sampleFrequency` are quite limited (depends on the possible timespan). 
I add this check is to prevent the wrong inputs.

Reply:
> For me, this function is **only** useful for the graph.

Mmmh. To me, this function is only **used** by the current charts we have, which only have some particular values as input. However, it could be useful for other charts using other sample sizes/steps, no?

> And the choices of sampleDuration and sampleFrequency are quite limited (depends on the possible timespan).
> I add this check is to prevent the wrong inputs.

I understand that we only give a user certain choices, through the selection of a timespan. But why would another choice be **wrong** for this utility function?

Let's take another example, coming from the real world. Imagine, a common tool called a screwdriver. Imagine you're building a screwdriver, so you can provide it to your users for, you guessed it... driving some screws. Now, let's say you only give 30mm, 40mm, and 50mm screws to your users, to use with your screwdriver. Does it mean the screwdriver **cannot** drive 60mm screws? Should the screwdriver break if it were given such a screw, by design?

Reply:
Thanks for the explanation! 

I agree! Since it's a utility function we should not limit the choices. 

We can still check the properties `sampleFrequency` and `sampleDuration` if they're based on the choices we provided. But the checking should be at where we call the function.